File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepageContract.java
package com.wayfair.themoviedb.feature.homepage;

import com.pubbix.base.BaseView;
import com.wayfair.themoviedb.base.BaseRouter;

public interface HomepageContract {
    interface View extends BaseView {
        void setOnQueryTextListener();
    }

    interface Router extends BaseRouter {
        void goToSearchPage(String searchTerms);
    }

    interface Presenter {
        void setUpViews();

        void search(String searchTerms);

        void setInteractor(Interactor interactor);
    }

    interface Interactor {
        void setPresenter(Presenter presenter);

    }

    interface Repository{
        void setInteractor(Interactor interactor);
    }
}

File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepageContract.kt
package com.wayfair.themoviedb.feature.homepage

import com.wayfair.themoviedb.base.BaseRouter
import com.wayfair.themoviedb.base.BaseView

interface HomepageContract {
    interface View : BaseView

    interface Router : BaseRouter {
        fun goToSearchPage(searchTerms: String)
    }

    interface Presenter {

        fun search(searchTerms: String)
    }

    interface Interactor {
        fun setPresenter(presenter: Presenter)

    }

    interface Repository {
        fun setInteractor(interactor: Interactor)
    }
}
